core/spec/models/spree/promotion_code_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe Spree::PromotionCode do
+
+  describe "normalize blank values for code & path" do
+    let(:code) { create(:promo_code, value: '') }
+
+    it "will save blank value as nil value instead" do
+      expect(code.value).to be_nil
+    end
+  end
+
+  describe "expiry dates" do
+    let(:code) { create(:promo_code) }
+
+    pending "is invalid before starts_at"
```

Comment:
should finish these specs if we allow expiration on a code basis rather than on the promotion itself